scripts/run_frontend_tests_test.py

```diff
@@ -44,76 +44,110 @@ def mock_print(msg: str, end: str = '\n') -> None:  # pylint: disable=unused-arg
             self.print_arr.append(msg)
         self.print_swap = self.swap(builtins, 'print', mock_print)
 
-        class MockFile:
-            def __init__(self, flakes: int = 0) -> None:
-                self.counter = 0
+        class MockStdout:
+            def __init__(
+                self,
+                flakes: int = 0,
+                counter: int = 0,
+                print_karma_logs: bool = False
+            ) -> None:
+                """Mock stdout for testing.
+
+                Args:
+                    flakes: int. The number of times to re-run the task.
+                        Defaults to 0.
+                    counter: int. The number of lines that have been read.
+                        Defaults to 0.
+                    print_karma_logs: bool. Whether to simulate printing a log
+                        from the karma web-server for testing that those logs
+                        are ignored. Defaults to False.
+                """
+                self.counter = counter
                 self.run_counter = 0
                 self.flakes = flakes
+                self.print_karma_logs = print_karma_logs
+
+            def readline(self) -> bytes:
+                """Simulates reading lines from stdout.
 
-            def readline(self) -> bytes: # pylint: disable=missing-docstring
+                Each invocation reads a new line, in the process incrementing
+                counter. When readline returns empty bytes, the process is no
+                longer printing to stdout.
+
+                Returns:
+                    bytes. The line that has been read.
+                """
                 self.counter += 1
                 if self.counter == 1:
                     return (
                         b'Executed tests. Trying to get the Angular injector..')
+                if self.counter == 2 and self.print_karma_logs:
```

Comment:
Add a comment above this line saying something like "Karma logs are prefixed with [web-server]".